src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsallocations/service/TriggerReprocessService.kt

```diff
@@ -0,0 +1,55 @@
+package uk.gov.justice.digital.hmpps.hmppsallocations.service
+
+import com.amazonaws.services.sqs.AmazonSQS
+import com.amazonaws.services.sqs.model.MessageAttributeValue
+import com.amazonaws.services.sqs.model.SendMessageRequest
+import com.amazonaws.services.sqs.model.SendMessageResult
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.scheduling.annotation.Async
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsallocations.listener.HmppsOffenderEvent
+import uk.gov.justice.digital.hmpps.hmppsallocations.listener.SQSMessage
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.MissingQueueException
+
+@Service
+class TriggerReprocessService(
+  private val hmppsQueueService: HmppsQueueService,
+  private val objectMapper: ObjectMapper,
+  @Qualifier("hmppsoffenderqueue-sqs-client") private val hmppsOffenderSqsClient: AmazonSQS,
+) {
+
+  private val hmppsOffenderQueueUrl = hmppsQueueService.findByQueueId("hmppsoffenderqueue")?.queueUrl ?: throw MissingQueueException("HmppsQueue hmppsoffenderqueue not found")
+
+  @Async
```

Comment:
If we are using coroutines then we ought to use async blocks here - like the lookup service